clientapi/routing/sendevent.go

```diff
@@ -80,6 +81,29 @@ func SendEvent(
 		return *resErr
 	}
 
+	stateRes := api.QueryLatestEventsAndStateResponse{}
+	if err := rsAPI.QueryLatestEventsAndState(req.Context(), &api.QueryLatestEventsAndStateRequest{
+		RoomID:       roomID,
+		StateToFetch: []gomatrixserverlib.StateKeyTuple{},
+	}, &stateRes); err != nil {
+		util.GetLogger(req.Context()).WithError(err).Error("queryAPI.QueryLatestEventsAndState failed")
+		return jsonerror.InternalServerError()
+	}
+
+	// Check that currnet event doesn't match previous events
```

Comment:
I'm not really sure why we would want to do this? It doesn't really seem related to the fixes at hand.

Not to mention that the `QueryLatestEventsAndState` operation is an *expensive* thing to be calling on every sent event.

Reply:
This could easily not be the most optimal way to check for previous events, first blush at hacking on Dendrite. It was meant to be an attempt at making setting state idempotent by sending the previous identical event rather than sending a newly created event. 

Not sure if there is a better place in the code base to check for this, or a less expensive way to do so.

Reply:
I'm looking at #1329, what would be the recommended fix for this?